src/utils/wallet.js

```diff
@@ -310,6 +310,26 @@ class Wallet {
         });
     }
 
+    async deleteRecoveryMethod(method) {
+        await sendJson('POST', `${ACCOUNT_HELPER_URL}/account/deleteRecoveryMethod`, {
+            accountId: wallet.accountId,
+            recoveryMethod: method.kind,
+            publicKey: method.publicKey,
+            ...(await wallet.signatureFor(wallet.accountId))
+        }).then(({ error }) => {
```

Comment:
why use `await` and `.then` together?  Also `{ error }` would only be returned for promises returned from `dispatch`, stuff not wrapped into Redux actions would just throw error.